pages/price-feeds/use-real-time-data/aptos.mdx

```diff
@@ -1,31 +1,87 @@
 
-## Contract Addresses
+```toml copy
+[addresses]
+pyth = "0x7e783b349d3e89cf5931af376ebeadbfab855b3fa239b7ada8f5a92fbea6b387"
+deployer = "0xb31e712b26fd295357355f6845e77c888298636609e93bc9b05f0f604049f434"
+wormhole = "0x5bc11445584a763c1fa7ed39081f1b920954da14e04b32440cba863d03e19625"
+```
 
-Developers will need the address of the Pyth price feed contract on their blockchain in order to use Pyth.
-Please consult [Aptos Contract Addresses](../contract-addresses/aptos) to find the address for your blockchain.
+Consult: [Aptos Contract Addresses](../contract-addresses/aptos) for the complete list of contract addresses on different Aptos networks.
 
-## Price Feed IDs
+## Write Contract Code
 
-| Network       | Available Price Feeds                                                                                                        |
-| ------------- | ---------------------------------------------------------------------------------------------------------------------------- |
-| Aptos Testnet | [https://pyth.network/developers/price-feed-ids#aptos-testnet](https://pyth.network/developers/price-feed-ids#aptos-testnet) |
-| Aptos Mainnet | [https://pyth.network/developers/price-feed-ids#aptos-mainnet](https://pyth.network/developers/price-feed-ids#aptos-mainnet) |
+The code snippet below provides an example module fetching the BTC/USD price from Pyth price feeds:
+
+```rust {21} copy
+module example::example {
+    use pyth::pyth;
+    use pyth::price::Price;
+    use pyth::price_identifier;
+    use aptos_framework::coin;
+
+    // Fetch the pyth_update_data from Hermes and pass it to the Pyth contract to update the prices.
+    // Add the pyth_update_data argument to any method on your contract that needs to read the Pyth price.
+    // See https://docs.pyth.network/price-feeds/fetch-price-updates for more information on how to fetch the pyth_update_data.
+    public fun get_btc_usd_price(user: &signer, pyth_update_data: vector<vector<u8>>): Price {
+
+        // First update the Pyth price feeds
+        let coins = coin::withdraw(user, pyth::get_update_fee(&pyth_update_data));
+        pyth::update_price_feeds(pyth_update_data, coins);
+
+        // Read the current price from a price feed.
+        // Each price feed (e.g., BTC/USD) is identified by a price feed ID.
+        // The complete list of feed IDs is available at https://pyth.network/developers/price-feed-ids
+        let btc_price_identifier = x"e62df6c8b4a85fe1a67db44dc12de5db330f7ac66b72dc658afedf0f4a415b43";
+        let btc_usd_price_id = price_identifier::from_byte_vec(btc_price_identifier);
+        pyth::get_price(btc_usd_price_id)
+    }
+}
+
+```
+
+<Callout type="info" emoji="ℹ️">
+  The `pyth_update_data` argument is required to update the Pyth price feeds.
+  This data contains the price updates fetched from Hermes and passed to the
```

Comment:
@jayantk I feel this sentence is too big. Should we rephrase it. 🤔

Reply:
How about:

The `pyth_price_update` argument contains verified prices from Pyth. Calling `pyth::update_price_feeds` with this value updates the on-chain Pyth price, ensuring your application has recent price data. The `pyth_price_update` can be fetched from Hermes; see [How to Fetch Price Updates] for more information.